src/main/java/org/springframework/data/elasticsearch/client/reactive/DefaultReactiveElasticsearchClient.java

```diff
@@ -177,8 +190,24 @@ public static ReactiveElasticsearchClient create(HttpHeaders headers, String...
 	 * @return new instance of {@link DefaultReactiveElasticsearchClient}.
 	 */
 	public static ReactiveElasticsearchClient create(ClientConfiguration clientConfiguration) {
+		return create(clientConfiguration, new DefaultRequestCreator());
+	}
+
+	/**
+	 * Create a new {@link DefaultReactiveElasticsearchClient} given {@link ClientConfiguration} and
+	 * {@link RequestCreator}. <br />
+	 * <strong>NOTE</strong> If the cluster requires authentication be sure to provide the according {@link HttpHeaders}
+	 * correctly.
+	 *
+	 * @param clientConfiguration Client configuration. Must not be {@literal null}.
+	 * @param requestCreator Request creator. Must not be {@literal null}.
+	 * @return new instance of {@link DefaultReactiveElasticsearchClient}.
+	 */
+	public static ReactiveElasticsearchClient create(ClientConfiguration clientConfiguration,
```

Comment:
‼️ cannot comment on the line 216 where the `DefaultReactiveElasticsearchClient(hostProvider)`call is made, but there the `requestCreator` constructor argument is missing.

Reply:
Oops, that's embarassing. Fixed.